core/engine/src/object/builtins/jstypedarray.rs

```diff
@@ -273,6 +358,104 @@ impl JsTypedArray {
         )
     }
 
+    /// Calls `TypedArray.prototype.findIndex()`.
+    #[inline]
+    pub fn find_index(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<Option<u64>> {
+        let index = BuiltinTypedArray::find_index(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )?
+        .as_number()
+        .expect("TypedArray.prototype.findIndex() should always return number");
+
+        if index >= 0.0 {
+            Ok(Some(index as u64))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Calls `TypedArray.prototype.findLast()`.
+    #[inline]
+    pub fn find_last(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        BuiltinTypedArray::find_last(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )
+    }
+
+    /// Calls `TypedArray.prototype.findLastIndex()`.
+    #[inline]
+    pub fn find_last_index(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<Option<u64>> {
+        let index = BuiltinTypedArray::find_last_index(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )?
+        .as_number()
+        .expect("TypedArray.prototype.findLastIndex() should always return number");
+
+        if index >= 0.0 {
+            Ok(Some(index as u64))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Calls `TypedArray.prototype.forEach()`.
+    #[inline]
+    pub fn foreach(
```

Comment:
nit: Should this be in snake case?

This is extending the API from `BuiltinTypedArray`, but we use `for_each` in other places in the engine. Would it be better to maintain consistency?

Reply:
Hi @nekevss, big thanks for your support! Apologies for the delay; I got tied up with something else. Now, I'm eager to maintain continuity with this.

I used `foreach` because the internal API `BuiltinTypedArray::foreach` uses it too. But I'm flexible to rename it for consistency with our preferred naming conventions, following your guidance.

Reply:
@jedel1043 any thoughts on the use of snake case with `foreach` on the internal API for `BuiltinTypedArray`.

In general, I'm of the opinion for at least the external API to be closer to the `Rust` convention. Anyone who consumes the API will probably expect snake case. At minimum, we should probably be at least consistent throughout the engine on whether it's `foreach` or `for_each`.

Reply:
I think we mostly agree on always converting camelCase JS APIs to snake_case Rust APIs. IMO, functions that differ from this convention should be treated as naming bugs.

Reply:
Sounds good! 😄 Then let's go with renaming the functions from `foreach` to `for_each` for naming.